### DISCUSSION TITLE
Tag and Rule installation

### DISCUSSION THREAD

**Discussion by tychobrailleur** [green-code-initiative/ecoCode#357] [Q&A]
Hi All,

A while back I successfully managed to deploy and run ecoCode on a local SonarQube instance, following this [guide](https://github.com/green-code-initiative/ecoCode-common/blob/main/doc/HOWTO.md#installing-local-environment-local-sonarqube).  The creation of tags & rules is done through bash scripts. 

Now I am wondering about installation through the SonarQube marketplace, which I believe would be the normal way to deploy in a “Production” environment.  While the installation of the plugins seems to complete successfully, I cannot find any documentation on how a SonarQube administrator can deploy ecoCode rules and tags.  Are they expected to run the scripts above?  This may not be possible/allowed in various Production environments.

Could someone point me to instructions?  It doesn't seem immediately obvious to me.  Thanks!

**Comment by dedece35** [comment]
Hi @tychobrailleur,

thank you for question !
Maybe you have seen thath "ecocode" has been renamed to "creedengo".

Indeed, we wrote a documentation for installation of our plugins rules in a local Sonarqube instance because it is installed for a developper in a local machine.
If we want to install plugin rules in an existing SonarQube instance, we shouldn't install rules automatically and it isn't a good practice for us.
If we check how SonarQube plugin system works : 
- we can add a plugin to a SonarQube instance with the MarketPlace
- next, a new behaviour (with some rules maybe) is added to SonarQube instance but administrator controls this behaviour.

For our plugins, once a plugin is installed into your SonarQube instance, all rules of this plugin are available : you can see them if you go to "rules" tab and display rules with "creedengo" tag.
But, by default, no rules are automatically enabled because we don't want to change your SonarQube behaviour.

As an SonarQube administrator, you can enable some rules in different ways : 
- create a new profile with only Creedengo rules and use this profile for some analyzed projects
- edit an existing profile and add all Creedengo rules you want
The next analysis of projects using this updated profile, will check Creedengo rules.

But you're right, we d'ont have a documentation that explains it.
But maybe, can you contribute and write a new section in our "HOWTO" documentation based on your experience ?

Thank you.
